File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    cdict = {}
    for clothe in clothes:
        name, genre = clothe
        if genre not in cdict:
            cdict[genre] = [name]
        else:
            cdict[genre].append(name)
    tlst = []
    for key in cdict:
        answer *= len(cdict[key])+1
    answer -=1 
            
    return answer

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
using namespace std;

int solution(vector<vector<string>> clothes) {
    map<string,int> mp;
    
    for (auto c : clothes) {
        std::cout << c[0] << " : " << c[1] << endl;
        mp[c[1]]++;
    }
    int answer = 1;
    for (auto pair : mp){
        std::cout << pair.second << endl;

        answer *= (pair.second+1);
    }
    return answer-1;
}